### PR TITLE
Imgutil adding tests, test utils and refactor

### PR DESCRIPTION
**Pull request by dlion**
RFC: https://github.com/buildpacks/rfcs/blob/main/text/0111-support-insecure-registries.md

### CODE REVIEW

File: remote/options.go [hunk 1]
```diff
@@ -99,6 +99,7 @@ func WithPreviousImage(imageName string) ImageOption {
 // the image. The referenced images could include the base image, a previous image, or the image itself.
 func WithRegistrySetting(repository string, insecure, insecureSkipVerify bool) ImageOption {
 	return func(opts *options) error {
+		opts.registrySettings = make(map[string]registrySetting)
```
Comment (dlion) [inline remote/options.go:102]: This fixes a bug that panic for missing initialization of the map

**Comment by natalieparellano** [comment]
@dlion is it possible to resolve the merge conflict?

**Comment by dlion** [comment]
@natalieparellano done 😄

**Comment by dlion** [comment]
@natalieparellano is there anything else I need to do on this PR to get merged?

**Review by natalieparellano — APPROVED** [review]

**Pull request merged**